### PR TITLE
[minigraph.py] generate port description for every port

### PR DESCRIPTION
**Pull request by mykolaf**
Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

**- What I did**
Add logic to always generate port description for PORT. The goal is to provide the description in one place for snmp, lldp and user. (previously lldpmgr generated its own port description)
**- Why I did it**
My goal is for each port to have some description. The concern here is that currently lldpd will advertise SONiC ports with some default description of it's own, not matching the one in DB, used by snmp & shown to user.

We will the description generated in one place and will configure the lldpd accordingly. https://github.com/Azure/sonic-buildimage/pull/2396

**- How I did it**
The logic is the following:
If the port description is specified in port_config.ini/minigraph - use it.
If the minigraph has info about the neighbor for this port - create description in form "{neighbor_host}:{neighbor_port}"
Else assign it to port name.
**- How to verify it**
sonic-cfggen -m /etc/sonic/minigraph.xml -p  /usr/share/sonic/device/x86_64-mlnx_msn2100-r0/ACS-MSN2100/port_config.ini --print-data

port_config.ini
```
# name          lanes         speed description
Ethernet0       0             10000   q
Ethernet1       1             10000
Ethernet2       2             10000   w
```
```
Result :
    "PORT": {
        "Ethernet0": {
            "lanes": "0",
            "alias": "Ethernet0",
            "speed": "10000",
            "description": "q",
            "mtu": "9100"
        },
        "Ethernet1": {
            "lanes": "1",
            "description": "Servers0:eth0",
            "mtu": "9100",
            "alias": "Ethernet1",
            "admin_status": "up",
            "speed": "10000"
        },
        "Ethernet2": {
            "lanes": "2",
            "description": "w",
            "mtu": "9100",
            "alias": "Ethernet2",
            "admin_status": "up",
            "speed": "10000"
 . . .
        "Ethernet36": {
            "lanes": "36,37,38,39",
            "description": "ARISTA04T1:Ethernet1",
            "mtu": "9100",
            "alias": "Ethernet36",
            "admin_status": "up",
            "speed": "50000"
        },
        "Ethernet40": {
            "alias": "Ethernet40",
            "lanes": "40,41,42,43",
            "description": "Ethernet40",
            "speed": "40000",
            "mtu": "9100"
        }
```
**- Description for the changelog**

generate mandatory default port description

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -530,13 +530,26 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
 
         ports.setdefault(port_name, {})['description'] = port_descriptions[port_name]
 
+    # for the ports w/o description try to set one based on neighbor info, or just port name by default
+    for port_name, port in ports.items():
+        if not port.get('description'):
+            try:
+                neigh_host = neighbors[port_name]['name']
+                neigh_port = neighbors[port_name]['port']
+                port['description'] = neigh_host + ':' + neigh_port
```
Comment (nikos-github) [inline src/sonic-config-engine/minigraph.py]: I don't agree with your change here. Port description shouldn't be coming from port_config.ini and certainly shouldn't default to neigh_host + neigh_port.

Comment (mykolaf) [reply src/sonic-config-engine/minigraph.py]: This is not encouraging to specify port description via port_config.ini.

 As for default value I'd be glad to hear some suggestions.
My concern is not matching the descr advertised by lldp and snmp. The neighbor info and port name is the defaults that lldp is using. 

Let's start a discussion

Comment (mykolaf) [reply src/sonic-config-engine/minigraph.py]: Hi @nikos-github , can you comment/approve this?

**Comment by yxieca** [comment]
Made to 201811 branch on 2/14/2019

**Pull request merged**